File: backend/app/models/account.py
from __future__ import annotations

import enum
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base
from typing import Optional


class Platform(str, enum.Enum):
    wechat_mp = "wechat_mp"
    xiaohongshu = "xiaohongshu"
    douyin = "douyin"
    wechat_video = "wechat_video"


class AccountStatus(str, enum.Enum):
    active = "active"
    inactive = "inactive"
    error = "error"


class Account(Base):
    __tablename__ = "accounts"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False)
    platform: Mapped[Platform] = mapped_column(String(20), nullable=False)
    nickname: Mapped[str] = mapped_column(String(200), nullable=False)
    avatar_url: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    status: Mapped[AccountStatus] = mapped_column(
        String(20), default=AccountStatus.active, nullable=False
    )
    cookie_data: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    access_token: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    token_expires_at: Mapped[Optional[datetime]] = mapped_column(DateTime, nullable=True)
    last_check_at: Mapped[Optional[datetime]] = mapped_column(DateTime, nullable=True)
    error_message: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now(), nullable=False
    )

    user = relationship("User", back_populates="accounts")
    publish_tasks = relationship(
        "PublishTask", back_populates="account", cascade="all, delete-orphan"
    )

File: backend-go/models/account.go
package models

import (
	"time"
)

const (
	PlatformWechatMP    = "wechat_mp"
	PlatformXiaohongshu = "xiaohongshu"
	PlatformDouyin      = "douyin"
	PlatformWechatVideo = "wechat_video"

	AccountStatusActive   = "active"
	AccountStatusInactive = "inactive"
	AccountStatusError    = "error"
)

type Account struct {
	ID             string     `orm:"column(id);pk;size(36)" json:"id"`
	UserID         string     `orm:"column(user_id);size(36);index" json:"user_id"`
	Platform       string     `orm:"column(platform);size(20)" json:"platform"`
	Nickname       string     `orm:"column(nickname);size(200)" json:"nickname"`
	AvatarURL      string     `orm:"column(avatar_url);size(500);null" json:"avatar_url"`
	Status         string     `orm:"column(status);size(20);default(active)" json:"status"`
	CookieData     string     `orm:"column(cookie_data);type(text);null" json:"cookie_data"`
	AccessToken    string     `orm:"column(access_token);type(text);null" json:"access_token"`
	TokenExpiresAt *time.Time `orm:"column(token_expires_at);type(datetime);null" json:"token_expires_at"`
	LastCheckAt    *time.Time `orm:"column(last_check_at);type(datetime);null" json:"last_check_at"`
	ErrorMessage   string     `orm:"column(error_message);type(text);null" json:"error_message"`
	CreatedAt      time.Time  `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
	UpdatedAt      time.Time  `orm:"column(updated_at);auto_now;type(datetime)" json:"updated_at"`
}

func (a *Account) TableName() string {
	return "accounts"
}
